hf_olmo/convert_olmo_to_hf.py

```diff
@@ -60,46 +101,157 @@ def write_tokenizer(checkpoint_dir: str):
 
 
 def convert_checkpoint(checkpoint_dir: str, ignore_olmo_compatibility: bool = False):
+
+    print("Converting checkpoint to HF format...")
     write_config(checkpoint_dir)
+
+    print("Saving model to checkpoint...")
     write_model(checkpoint_dir, ignore_olmo_compatibility=ignore_olmo_compatibility)
+
+    print("Saving tokenizer to checkpoint...")
     write_tokenizer(checkpoint_dir)
 
     # Cannot remove it before writing the tokenizer
     if ignore_olmo_compatibility:
         os.remove(os.path.join(checkpoint_dir, "config.yaml"))
 
 
-def download_remote_checkpoint_and_convert_to_hf(checkpoint_dir: str, local_dir: str):
-    from cached_path import cached_path
+def fix_tokenizer(checkpoint_dir: str, tokenizer_name_or_path: Optional[str] = None):
+    path = os.path.join(checkpoint_dir, "config.yaml")
+    conf = om.load(path)
 
-    model_name = os.path.basename(checkpoint_dir)
-    local_model_path = os.path.join(local_dir, model_name)
-    os.makedirs(local_model_path, exist_ok=True)
+    print("Saving tokenizer to checkpoint...")
 
-    model_files = ["model.pt", "config.yaml"]  # , "optim.pt", "other.pt"]
-    for filename in model_files:
-        final_location = os.path.join(local_model_path, filename)
-        if not os.path.exists(final_location):
-            remote_file = os.path.join(checkpoint_dir, filename)
-            logger.debug(f"Downloading file {filename}")
-            cached_file = cached_path(remote_file)
-            shutil.copy(cached_file, final_location)
-            logger.debug(f"File at {final_location}")
-        else:
-            logger.info(f"File already present at {final_location}")
+    tokenizer_name_or_path = str(tokenizer_name_or_path or conf["tokenizer"]["identifier"])  # pyright: ignore
 
-    convert_checkpoint(local_model_path)
-    return local_model_path
+    try:
+        Tokenizer.from_pretrained(tokenizer_name_or_path)
```

Comment:
I wanna be able to run this script outside the OLMo directory (that is, just after installing ai2-olmo via pip), and in the case I wouldn't have json vocabulary file in path (not particularly fond of those being in the olmo repo anyway). 

gonna re-giggle this a bit to support both.

Reply:
I changed `Tokenizer.from_checkpoint` to work with pip-installed ai2-olmo (in particular, I handled the scenario where the tokenizer field is something like `tokenizers/allenai_eleuther-ai-gpt-neox-20b-pii-special.json`). If it doesn't work in that scenario, please let me know so that I can fix it as some point.